Add default file inclusion support in .NET SDK for RESW and image files in WinAppSDK .NET projects.

rohanp-msft opened the pull request:
### Description
This is for the feature request: https://github.com/dotnet/sdk/issues/24093
Please refer to it for a description of the change.

### How verified
#### Automated
Added tests.

#### Manual
Tested these cases for WinUI, packaged console, packaged WPF and packaged WinForms:
1. New RESW file
2. RESW file added via UI from external folder
3. PNG file added via UI from external folder
4. RESW file copy pasted into project folder
5. JPG file copy pasted into project folder

Note: WPF + RESW doesn't work. The attempted inclusion is undone by automatic injection in the project file. By that, I mean that this is injected: removal from PRIResource and addition to None. That needs to be fixed separately, likely in the Windows App SDK. Tracked by https://github.com/microsoft/WindowsAppSDK/issues/2324.

Note 2: Tested with and without https://github.com/microsoft/WindowsAppSDK/blob/main/dev/MRTCore/packaging/MrtCore.props in the Windows App SDK NuGet package. So, that WAS change and this can coexist! In the long term though, the WAS change should be removed. I also confirmed that the string and image resources were not showing up twice in the PRI file.

Description of setup:
Used VS2022.
Modified .NET SDK install in C:\Program Files\dotnet\sdk\.

rohanp-msft committed:
Add default file inclusion support in .NET SDK for RESW and image files in WinAppSDK .NET projects

huichen123 commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props:
```diff
@@ -28,15 +28,24 @@ Copyright (c) .NET Foundation. All rights reserved.
   <ItemGroup Condition=" '$(EnableDefaultItems)' == 'true' ">
     <Compile Include="**/*$(DefaultLanguageSourceExtension)" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" Condition=" '$(EnableDefaultCompileItems)' == 'true' " />
     <EmbeddedResource Include="**/*.resx" Exclude="$(DefaultItemExcludes);$(DefaultExcludesInProjectFolder)" Condition=" '$(EnableDefaultEmbeddedResourceItems)' == 'true' " />
+    <!-- WindowsAppSdk is a NuGet delivered Windows SDK. Include only if the package is installed. -->
```
"Windows SDK" is a different thing

rohanp-msft replied:
How about "app SDK"? If that's not OK, what do you think is best?

huichen123 replied:
just SDK?

dsplaisted replied:
The condition is based on the `WindowsAppSdkFoundation` property, not based on the NuGet package being referenced as the comment seems to suggest.  Does the NuGet package set the property?  Or is a project referencing the NuGet package expected to set the property?

It would be good to clarify the comment.

dsplaisted commented:
/azp run

dsplaisted commented on src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.props:
```diff
@@ -28,15 +28,24 @@ Copyright (c) .NET Foundation. All rights reserved.
   </PropertyGroup>
 
+  <PropertyGroup>
+    <_ImageGlobs>**/*.png;**/*.bmp;**/*.jpg;**/*.dds;**/*.tif;**/*.tga;**/*.gif</_ImageGlobs>
```
I'd suggest changing this property name to `_WindowsAppSdkDefaultImageIncludes`.

dsplaisted reviewed (approved):
It would be nice to have a test covering this logic.  You can see some existing tests covering other item includes here: https://github.com/dotnet/sdk/blob/main/src/Tests/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs

It's up to you whether you want to add the test though.  I'm OK taking the PR as is.  A test would reduce the risk that this functionality would break.

rohanp-msft committed:
Addressed comments

rohanp-msft committed:
Added tests.

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=24139) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=24139) before we can accept your contribution.<br/><br/>:x: rohanp-msft [sign now](https://cla.dotnetfoundation.org/dotnet/sdk?pullRequest=24139)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/sdk?pullRequest=24139) it.</sub>

rohanp-msft commented:
/azp run

rohanp-msft commented:
@dsplaisted Daniel, I added tests per your suggestion and addressed your and Hui's comments. Would you please merge the pull request? I don't have permission to do it.

rohanp-msft commented:
> @dsplaisted Daniel, I added tests per your suggestion and addressed your and Hui's comments. Would you please merge the pull request? I don't have permission to do it.

@dsplaisted On second thought, I'll take some more time to make sure I didn't miss anything. I don't think I did... but better safe than sorry. Will update when ready. Sorry for the churn.
